lib/string_reverse.rb

```diff
@@ -1,6 +1,17 @@
 # A method to reverse a string in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(1)
+
 def string_reverse(my_string)
-  raise NotImplementedError
+  return nil if my_string == nil
+
+  length = my_string.length
+
+  (length / 2).times do |i|
+    temp_i = my_string[my_string.length - (i + 1)]

```

Comment:
Nit: `my_string.length - (i + 1)` is probably best moved into its own variable for two reasons:
* it's used multiple times
* it's not _immediately_ clear what this represents - maybe `swap_index`? (I had to glance at it for a second or two.)